### PR TITLE
Storage-ng based AutoYaST partitioning cloning (rebased)

### PR DESCRIPTION
**Pull request by ancorgs**
This replaces #280 (a rebase was needed)

Part of https://trello.com/c/mAFPlMj7/602-5-autoyast-ng-partitioner-clone-the-system-basic

Used from https://github.com/yast/yast-autoinstallation/pull/311

### CODE REVIEW

File: test/autoinst_profile/drive_section_test.rb [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    it "initializes #type to :CT_DISK for both disks and DASDs" do
+      expect(described_class.new_from_storage(device("dasdb")).type).to eq :CT_DISK
+      expect(described_class.new_from_storage(device("sdc")).type).to eq :CT_DISK
+    end
+
+    describe "initializing DriveSection#device" do
+      let(:dev) { device("sdd") }
+
+      before do
+        expect(Yast::Arch).to receive(:s390).and_return s390
```
Comment (imobachgs) [inline test/autoinst_profile/drive_section_test.rb]: `expect` in a `before` block.

File: test/autoinst_profile/partition_section_test.rb [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+      it "initializes #loop_fs and #crypt_key to true" do
+        section = section_for("sdf7")
+        expect(section.crypt_fs).to eq true
+        expect(section.loop_fs).to eq true
+      end
+    end
+
+    context "if the partition is not encrypted" do
+      it "initializes #crypt_key to a generic string" do
```
Comment (imobachgs) [inline test/autoinst_profile/partition_section_test.rb]: Just curiosity: why not to `nil` or an empty string? I guess it's something from the past...

Comment (ancorgs) [inline test/autoinst_profile/partition_section_test.rb]: Yes, honoring legacy behavior. I will add a comment.

File: test/autoinst_profile/drive_section_test.rb (hunk 1 above)
Comment (ancorgs) [inline test/autoinst_profile/drive_section_test.rb]: This should clearly be an `allow`

**Pull request merged**